Source/WebGPU/WGSL/Lexer.cpp

```diff
@@ -41,7 +41,7 @@ static unsigned isIdentifierStart(UChar character, const UChar* start, const UCh
         return 1;
 
     unsigned length = 1;
-    if (u_charType(*(start + length - 1)) == U_SURROGATE && static_cast<unsigned>(end - start) > length)
+    if (end > start + 1 && u_charType(*start) == U_SURROGATE)
```

Comment:
I'm assuming changing `u_charType(*(start + length - 1))` to `u_charType(*start)` is fine @tadeuzagallo ?

Reply:
"end > start + 1" is equivalent to "static_cast<unsigned>(end - start) > length"

Reply:
oh I just realized that `length = 1` is on the line immediately above